core/src/library.rs

```diff
@@ -19,6 +23,7 @@ pub struct MovieLibrary<'gc> {
     jpeg_tables: Option<Vec<u8>>,
     device_font: Option<Font<'gc>>,
     fonts: HashMap<FontDescriptor, Font<'gc>>,
+    vm_tendency: Option<AvmType>,
```

Comment:
This shouldn't be an Option; we are always guaranteed to have an AVM type, and we should determine the AVM version before any execution.